### DISCUSSION TITLE
Buffering messages during Network Outage

### DISCUSSION THREAD

**Discussion by b-ferd** [python-telegram-bot/python-telegram-bot#4177] [Q&A]
Is there any suggested, elegant way to buffer messages that could not be sent during a network outage, for later re-send when the network comes up again?

The naive, knee-jerk idea would be to write a wrapper function that is used instead of `application.bot.send_message()`, let's call it `buffered_send()`.
This would do a try-except on `application.bot.send_message()`, and in case that fails, set a "network down" flag and append the passed arguments to a list.
The list would be flushed (_try to send_) by `buffered_send()` as a first step, element by element, if not empty, or (when the "network down" flag it set) by a regulary triggered alert (`JobQueue`).

However I have the feeling I'm building a [Rube Goldberg machine](https://en.wikipedia.org/wiki/Rube_Goldberg_machine) here and that this scenario already might be foreseen in the Framework somewhere... ?

**Comment by Bibo-Joshi** [comment]
One idea that comes to mind is that you could implement a custom subclass of `BaseRequest`/`HTTPXRequest` that includes an automated retry-mechanism.
